zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/OriginalDataAgent.scala

```diff
@@ -12,31 +13,18 @@ import scala.concurrent.ExecutionContext
 
 class OriginalDataAgent(override val dbName: String,
                         override val schema: Schema,
+                        val initCardinality: Cardinality,
                         override val queryParser: IQLGenerator,
                         override val conn: IDataConn,
                         override val config: Config)(implicit ec: ExecutionContext)
   extends AbstractDataSetAgent(dbName, schema, queryParser, conn, config)(ec) {
 
   import OriginalDataAgent._
 
-  val lastCount: Cardinality = UnInitialCardinality
+  val lastCount: Cardinality = initCardinality
 
-  /**
-    * Ask for the initial stats when the Agent starts.
-    * Stats including: minTimeStamp, maxTimeStamp, cardinality
-    */
-  override def preStart(): Unit = {
-    val minTimeQuery = Query(dbName, globalAggr = Some(GlobalAggregateStatement(AggregateStatement(schema.timeField, Min, Field.as(Min(schema.timeField), "min")))))
```

Comment:
I think we still need the `preStart` to get the delta stats since last update time indicating in the `stats.getMaxTime` or something equivalent.